doc/BUILD.rst

```diff
@@ -0,0 +1,722 @@
+Taisei Project - Build/Maintainer FAQ
+=====================================
+
+This is intended for anyone looking to build or package Taisei Project
+for any of its supported operating systems, including Linux, macOS, Windows,
+Emscripten, Switch, and others.
+
+.. contents::
+
+Dependencies
+------------
+
+Taisei's mandatory dependencies are as follows:
+
+-  C (C11) compiler (``gcc``, ``clang``, etc)
+-  Python >= 3.5
+-  meson >= 0.56.2 (recommended)
+-  ninja >= 1.10.0
+-  docutils
+
+Built-In vs. System Dependencies
+""""""""""""""""""""""""""""""""
+
+Due to the wide array of platforms Taisei supports, the project packages many of
+its core dependencies inside the project using the
+`Meson dependency wrap system <https://mesonbuild.com/Wrap-dependency-system-manual.html>`__.
+This is used to pin specific versions of libraries such as `SDL2` to known-good
+versions, as well as assist with packaging and distribution on older and/or more
+esoteric systems (i.e: Emscripten).
+
+However, recompiling every single dependency Taisei needs every time you want to
+rebuild the project is not ideal. You can still use system dependencies
+installed through your OS's package manager (`apt`, `yum`, `brew`, etc) for
+development, local testing, or simply building the game for your own system.
+
+For convenience, ``meson`` will detect which packages are missing from your
+system and use its wrap dependency system to pull in what it can. The trade-off
+is that this can result in longer build times.
+
+For releases, however, you should rely *exclusively* on the ``meson`` wrap
+dependencies with the Taisei repository, for consistency and
+
+This is controlled through the ``--wrap-mode`` flag with ``meson``. (More
+on that later.)
+
+In summary:
+
+- Development = system-install dependencies are fine (`apt`, `brew`, etc)
+- Release = ``meson`` wrap dependencies only
```

Comment:
I don't think we should prescribe a dependency management strategy for people looking to package their own distributions. In the general case, system dependencies should be preferred, if the environment provides a sane package management mechanism. The reality is that few systems actually do, which is largely the reason why we have the subprojects for official releases, but that's not the only valid way to do it. A decent Linux distro package or a macOS homebrew formula would not use them, for example.